core/types/block.go

```diff
@@ -94,12 +83,21 @@ func (h *BlockHeader) BlockHash() hash.Hash {
 	// transactions.  Ignore the error returns since there is no way the
 	// encode could fail except being out of memory which would cause a
 	// run-time panic.
-	buf := bytes.NewBuffer(make([]byte, 0, MaxBlockHeaderPayload))
+	return h.Pow.GetBlockHash(h.BlockData())
+}
+
+// BlockHash computes the block identifier hash for the given block header.
+func (h *BlockHeader) BlockData() []byte {
```

Comment:
1.) the comment is not correct (`BlockHash computes ....` part)
2.) The method return the block header without ProofOfPow included. Where is the user of the Public method? if not please keep it private.